crates/autopilot/src/driver_model.rs

```diff
@@ -0,0 +1,124 @@
+//! Types for communicating with drivers.
+
+pub mod quote {
+    use model::u256_decimal;
+    use primitive_types::{H160, U256};
+    use serde::{Deserialize, Serialize};
+
+    #[derive(Clone, Debug, Default, Deserialize, Serialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct Request {
+        pub sell_token: H160,
+        pub buy_token: H160,
+        pub kind: Kind,
+        #[serde(with = "u256_decimal")]
+        pub amount: U256,
+    }
+
+    #[derive(Clone, Debug, Default, Deserialize, Serialize)]
+    #[serde(rename_all = "lowercase")]
+    pub enum Kind {
+        #[default]
+        Buy,
+        Sell,
+    }
+
+    #[derive(Clone, Debug, Deserialize, Serialize)]
+    #[serde(untagged, rename_all = "camelCase")]
+    pub enum Response {
+        Successful {
+            #[serde(with = "u256_decimal")]
+            sell_amount: U256,
+            #[serde(with = "u256_decimal")]
+            buy_amount: U256,
+            gas: u64,
+        },
+        Unfillable {
+            unfillable_reason: String,
+        },
+    }
+}
+
+pub mod solve {
+    use bigdecimal::num_bigint::BigUint;
+    use chrono::{DateTime, Utc};
+    use primitive_types::H160;
+    use serde::{Deserialize, Serialize};
+    use serde_with::{serde_as, DisplayFromStr};
+    use std::collections::BTreeMap;
+
+    #[derive(Clone, Debug, Default, Deserialize, Serialize)]
+    pub struct Request {
+        pub auction: Auction,
+        pub deadline: DateTime<Utc>,
```

Comment:
Is there a reason we split these two? Isn't `deadline` conceptually part of the `auction`?

Reply:
I like the `Request/Response` symmetry from all the modules - and am fond of the organization of the `driver_model`. Maybe:

```rust
pub struct Request {
    #[serde(flatten)]
    pub auction: Auction,
}

pub struct Auction {
    // ...
    deadline: DateTime<Utc>,
}
```

Reply:
I was following the openapi.  To me it doesn't fit better in either place, it doesn't matter.